### PR TITLE
Implement Multiversion Curation feature

### PR DESCRIPTION
**Pull request by disulliv**
This is the implementation of the feature proposal discussed in #809 . The design document has changes made during development and some screenshots of example curations as well.

All the code is behind a feature flag. We'd like to merge, turn it on in dev, then experiment to make sure everything is working as expected.

This is developed by myself and @MichaelTsengLZ . We're available for discussion about any questions or concerns.

### CODE REVIEW

File: lib/licenseMatcher.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      }
+    }
+    return {
+      isMatching: true,
+      match: compareResults.match
+    }
+  }
+}
+
+class DefinitionLicenseMatchPolicy {
```
Comment (nellshamrell) [inline lib/licenseMatcher.js:32]: This is excellent work!

File: providers/curation/github.js [hunk 2]
```diff
@@ -186,6 +256,19 @@ class GitHubCurationService {
     return serviceGithub.repos.createFile(fileBody)
   }
 
+  async _getUserInfo(githubCli) {
```
Comment (nellshamrell) [inline providers/curation/github.js:259]: Nice change!

**Review by nellshamrell — APPROVED** [review]
This is great work and I believe this is ready to be merged (especially since it is behind a feature flag). How about I merge this, you can activate it in the dev.clearlydefined.io environment, and then gather feedback in a separate GitHub issue? That way you can have an environment where it is currently working and solicit feedback from curators, etc.

Once we merge this to the dev environment and all the builds pass, I'm ok with merging this to production and keeping the feature hidden behind a flag until we feel it is good to go.

@disulliv @MichaelTsengLZ does that sound good to you?

**Comment by MichaelTsengLZ** [comment]
> This is great work and I believe this is ready to be merged (especially since it is behind a feature flag). How about I merge this, you can activate it in the dev.clearlydefined.io environment, and then gather feedback in a separate GitHub issue? That way you can have an environment where it is currently working and solicit feedback from curators, etc.
> 
> Once we merge this to the dev environment and all the builds pass, I'm ok with merging this to production and keeping the feature hidden behind a flag until we feel it is good to go.
> 
> @disulliv @MichaelTsengLZ does that sound good to you?

Sounds good to me.

**Comment by disulliv** [comment]
Sounds good to me. I just turned on the feature flag in dev as well, so it's all good to go.

**Pull request merged**